src/com/palmergames/bukkit/towny/listeners/TownyWorldListener.java

```diff
@@ -64,20 +66,38 @@ private void newWorld(World world) {
 				return;
 			}
 		}
-	
 
 		// This is a world we've never seen before, make a new TownyWorld.
 		TownyUniverse.getInstance().newWorld(world);
 		TownyWorld townyWorld = TownyAPI.getInstance().getTownyWorld(world.getUID());
-		if (townyWorld == null)
+		
+		if (townyWorld == null) {
 			TownyMessaging.sendErrorMsg("Could not create data for " + world.getName());
-		else if (world.getName().startsWith("DXL_") &&
+			return;
+		}
+		
+		TownyUniverse.getInstance().getDataSource().loadWorld(townyWorld);
+		
+		if (world.getName().startsWith("DXL_") &&
 				Bukkit.getServer().getPluginManager().getPlugin("DungeonsXL") != null) {
 			townyWorld.setUsingTowny(false);
 			townyWorld.save();
 		}
 	}
 
+	/**
+	 * Attaches UUIDs to worlds that might not have them
+	 */
+	private void checkWorlds() {

```

Comment:
This method probably should have the DXL test we do to exclude dungeons XL worlds, no?

Reply:
This just adds UUIDs to existing townyworlds that don't have one, when we create a townyworld for a DXL world it also has a uuid.